### ISSUE TITLE
SSE not working as expected

### ISSUE THREAD

**Issue by dami-izolan-axon** [kartikk221/hyper-express#261]
I couldn't make it work by using the example from [the docs](https://github.com/kartikk221/hyper-express/blob/master/docs/SSEventStream.md).

> Working version using regular Express down below

### Server code

```js
const HyperExpress = require("hyper-express");

const server = new HyperExpress.Server();

server.get("/sse", (request, response) => {
	if (response.sse) {
		console.log("client got connected");
		response.once("close", () => {
			console.log("client got disconnected");
		});
		
		response.sse.open();
		response.sse.send("sse connection open");

		setInterval(() => {
			response.sse.send("event sent");
		}, 5000);
	} else {
		response.send("SSE not supported");
	}
});

server.listen(3000, () => console.log("Listening on port 3000"));
```

#### TypeScript mismatch (seizing the opportunity)

The type of handler's `response` parameter is inferred as `Response<DefaultResponseLocals>`, which doesn't have an `sse` property, as opposed to `Response` class, which does have an `sse` property.

![image](https://github.com/kartikk221/hyper-express/assets/122882010/ec8c7354-cde6-427b-a421-3345373dd6e8)

### Postman screenshot

The client keeps waiting for a server response and never gets one – it's expected one event after each 5 seconds. 

![image](https://github.com/kartikk221/hyper-express/assets/122882010/9f5de759-363c-419c-b848-e98e37e4c56d)

### Server console output

Works as expected.

```
Listening on port 3000
client got connected
client got disconnected
```

### Server code

```js
const express = require("express");

const app = express();

app.get("/sse", (request, response) => {
	console.log("client got connected");
	request.on("close", () => {
		console.log("client got disconnected");
	});

	// setting up SSE manually
	response.writeHead(200, {
		"Content-Type": "text/event-stream",
		"Cache-Control": "no-cache",
		"Connection": "keep-alive",
	});
	response.write(`data: sse connection open\n\n`);

	setInterval(() => {
		response.write("data: event sent\n\n");
	}, 5000);
});

app.listen(3000, () => console.log("Listening on port 3000"));
```

### Postman screenshot

Postman recognizes this is an SSE endpoint and adapts the UI accordingly.

![image](https://github.com/kartikk221/hyper-express/assets/122882010/5dcfaf57-0edc-428b-95b3-901928a6bdb0)

### Server console output

Works the same.

```
Listening on port 3000
client got connected
client got disconnected
```

**Comment by kartikk221** [comment]
Could you post what headers are received by Postman and what the 84B received by Postman are from HyperExpress?

There is an extensive test for SSE here https://github.com/kartikk221/hyper-express/blob/master/tests/components/http/scenarios/response_sse.js which should not be passing If your code is broken so I suspect this may be something specific to either Postman or that the Node.js EventSource library may not be fully compatible with the spec for SSE.

**Comment by dami-izolan-axon** [comment]
Sure!

These are the response headers from HyperExpress:

![image](https://github.com/kartikk221/hyper-express/assets/122882010/294ed987-d66b-4e36-8f21-1e3cdb21763e)

The 84 bytes are the header's ones:

```
HTTP/1.1 200 OK
Date: Mon, 20 May 2024 20:37:26 GMT
Transfer-Encoding: chunked
```

By the way, I'm using **v6.16.1**.

I tried to make it work in the browser using JS's `EventSource` API but also had no success.

**Comment by kartikk221** [comment]
Hey, just published a new update with `v6.16.2` which should resolve your issue. It seems like the `response.sse.open()` method wasn't properly writing SSE headers but the Node.js Eventsource library used for testing just didn't throw an error on this.

**Comment by dami-izolan-axon** [comment]
**`v6.16.2`** has solved the problem.

Thank you, @kartikk221!
